### PR TITLE
[RDY] options: Enable &wildmenu, use "list:longest,full" for &wildmode by default

### PR DESCRIPTION
**Pull request by fmoralesc**
Re: #2676

### CODE REVIEW

**Comment by Pyrohh** [comment]
Heyho,

Regarding all of these PRs, please update vim_diff.txt and add a WIP or RFC tag (I don't know if you want them to be merged otherwise).

**Comment by fmoralesc** [comment]
Sure, will do.

File: src/nvim/option.c [hunk 1]
```diff
@@ -1624,10 +1624,10 @@ static vimoption_T
    {(char_u *)FALSE, (char_u *)0L} SCRIPTID_INIT},
   {"wildmenu",    "wmnu", P_BOOL|P_VI_DEF,
```
Comment (Pyrohh) [inline src/nvim/option.c]: Do we even want this to be defined as `P_VI_DEF`? I don't see a Vi default documented for them in `options.txt`. Same with `wildmode`.

File: runtime/doc/vim_diff.txt [hunk 2]
```diff
@@ -32,6 +32,8 @@ these differences.
 - 'encoding' defaults to "utf-8"
 - 'nocompatible' is always set
 - 'ttyfast' is always set
+- 'wildmenu' is enabled by default
```
Comment (Pyrohh) [inline runtime/doc/vim_diff.txt]: Maybe `set by default`, given Vim's notion of being set or unset.

File: src/nvim/option.c (hunk 1 above)
Comment (fmoralesc) [inline src/nvim/option.c]: :+1: Doh'

File: runtime/doc/vim_diff.txt (hunk 2 above)
Comment (fmoralesc) [inline runtime/doc/vim_diff.txt]: :+1:

**Comment by Pyrohh** [comment]
If the tests pass, then LGTM (commits should be squashed though).

**Comment by justinmk** [comment]
:+1:

**Comment by fmoralesc** [comment]
The [travis build](https://travis-ci.org/neovim/neovim/builds/62852101) failed, but only because of a timeout, so I'm squashing and marking as RDY.

**Comment by Pyrohh** [comment]
Sorry I didn't think of this earlier, but perhaps the tests should be modified and/or extended for the new defaults? That is, instead of just doing `set wildmode=full`.

**Comment by justinmk** [comment]
Merged

**Comment by justinmk** [comment]
> perhaps the tests should be modified and/or extended for the new defaults? That is, instead of just doing set wildmode=full.

In some cases that would be good, but for that specific test (which I wrote) it is irrelevant.

**Comment by jez** [comment]
Just curious (read: I've already added `set nowildmenu` to my ~/.nvimrc): is there a discussion of why this default was changed? It seems like a silly place to break compatibility with Vim. It makes no difference to me either way; I'm just a little confused as to _why_.

**Comment by fmoralesc** [comment]
@jez: `wildmenu` (I feel, and I should imagine others) provides a nicer interface for nvim's commandline, basically what menu completion gives to zsh and fish shell users (although a bit more limited). There are some arguments in favor of it [at vim-sensibe's issue tracker](https://github.com/tpope/vim-sensible/issues/81), which is what inspired us to use it.

**Comment by jez** [comment]
Ok, thanks. If you're going for the zsh/fish vibe, I would have thought `wildmode=list:full` to be closer to the default of these shells.

**Comment by fmoralesc** [comment]
For sure (I prefer `wildmode=full` myself, a list I can't navigate as in `wildmode=list` seems useless to me personally), but the consensus was on what we had here, and it at least had precedent in vim-sensible.

**Comment by justinmk** [comment]
@jez how does this break compatibility?

**Comment by jez** [comment]
It's not "compatibility" per se, but when I opened up nvim the other day and tried to tab complete, it was definitely jarring having never enabled `wildmenu` before.

**Comment by fmoralesc** [comment]
@jez Please check on the issue that tracks the new default proposed: #2676

**Pull request closed without merging**